core/src/main/resources/jenkins/security/ApiTokenProperty/config.jelly

```diff
@@ -26,132 +26,180 @@ THE SOFTWARE.
 <?jelly escape-by-default='true'?>
 <j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler" xmlns:f="/lib/form" xmlns:l="/lib/layout" xmlns:i="jelly:fmt" xmlns:d="jelly:define">
     <d:taglib uri="local">
-        <d:tag name="revokeIcon">
-          <l:icon src="symbol-trash" />
+        <d:tag name="editButton">
+            <button type="button"
+                    class="jenkins-table-button jenkins-button jenkins-button--tertiary api-token-property-token-rename"
+                    data-target-url="${descriptor.descriptorFullUrl}/rename" data-rename-title="${%RenameToken}"
+                    data-rename-message="${%New name}"
+                    tooltip="${%Rename this token}">
+                <l:icon src="symbol-edit"/>
+            </button>
+        </d:tag>
+        <d:tag name="revokeButton">
+            <button type="button"
+                    class="jenkins-table-button jenkins-button jenkins-button--tertiary jenkins-!-destructive-color api-token-property-token-revoke"
+                    data-message-if-legacy-revoked="${%RevokedToken}"
+                    data-confirm="${%ConfirmRevokeSingle}"
+                    data-confirm-title="${%ConfirmRevokeSingleTitle}"
+                    data-token-uuid="${token.uuid}"
+                    data-target-url="${descriptor.descriptorFullUrl}/revoke" tooltip="${%RevokeToken}">
+                <l:icon src="symbol-trash" />
+            </button>
         </d:tag>
     </d:taglib>
+    <template id="api-token-row-template" xmlns:local="local">
+        <tr>
+            <td>
+                <div class="token-name-container">
+                    <span class="token-name"/>
+                </div>
+            </td>
+            <td>
+                <span class="token-creation-date" tooltip="${%Just now}">${%Today}</span>
+            </td>
+            <td align="right" class="token-use-counter">0</td>
+            <td align="right">
+                <div class="token-last-used">${%Never}</div>
+            </td>
+            <td class="jenkins-table__cell--tight">
+                <div class="jenkins-table__cell__button-wrapper">
+                    <local:editButton/>
+                </div>
+            </td>
+            <td class="jenkins-table__cell--tight">
+                <div class="jenkins-table__cell__button-wrapper">
+                    <local:revokeButton/>
+                </div>
+            </td>
+        </tr>
+    </template>
+    <div class="jenkins-hidden" id="api-token-template">
```

Comment:
I'm not 100% sold on dialogs for showing the generated tokens.

Its possible to double click in the same place and never see the token.
Could we display it in the table temporarily?

Reply:
Could we disable the buttons for 200ms (or an appropriate amount of time) to avoid that?